do

```diff
@@ -1,2 +1,18 @@
 #!/bin/sh
-MAINJS=./node_build/make.js ./node_build/node.sh "$@"
+CARGO=$(command -v cargo)
+set -e
+if test "x$CARGO" = "x"; then
+    printf "Rust & Cargo are required in order to build cjdns\n"
+    printf "See https://rustup.rs/ for install instructions\n"
+    exit 1
+fi
+$CARGO build -vv --release
```

Comment:
Adding support for different build profiles (release vs debug) will be incredibly advantagous because differentiating debug builds opens the door to tons of build time saving optimizations.

Reply:
The reason I'm defaulting to always release builds is because the default way people install cjdns is by building the source with `./do` so that should make something like an installation result...

Reply:
Oh okay, I see. Your reasoning makes sense there.

However, for development builds, it might be nice to have the option of compiling a debug version. I see that the increased compile time is a serious pain point for you on this project, so compiling with a modified dev profile might help this process along. Is it okay if I include that in a merge against `rust-build` that I'm requesting?

I'll optimize it for the best compile time, obviously, but it will probably go something like this (inspired by [Bevy's fast compile strategy](https://github.com/bevyengine/bevy/blob/master/.cargo/config_fast_builds)).

```toml
[profile.dev]
lto = 'thin'
codegen-units = 48
rustflags = "-Zshare-generics=y" # would only work on nightly, so might not work
```

Additionally, using lld or zld as the linker will speed up link times (which [you also mentioned](https://mastodon.social/@cjd/104984414867630175) as a pain point XD)